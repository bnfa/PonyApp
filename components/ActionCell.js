import React from 'react';
import {TouchableOpacity, StyleSheet} from 'react-native';
import Cell from './Cell';

/**
 * Normal Cell component plus onPress function and Action
 * @param {array} borders Array of string containing ["norht", "west"]
 * @param {boolean} isDomokun true if the Cell index matches the Domokun position
 * @param {boolean} isEndPoint true if the Cell index matches the End-Point position
 * @param {boolean} isPony true if the Cell index matches the Pony position
 * @param {string} action can be 'west', 'east', 'north' , 'south' or 'stay'
 * @param {function} onPress update method for the direction of the pony
 */

function ActionCell({action, onPress, ...props}) {
  const isPony = action === 'stay' ? true : false;
  const color = isPony ? styles.pony : styles.neighbor;
  return (
    <TouchableOpacity
      onPress={() => {
        onPress();
      }}
      style={color}>
      <Cell {...props} isPony={isPony} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  pony: {
    backgroundColor: '#eebff2',
  },
  neighbor: {
    backgroundColor: '#f0f0f0',
  },
});
export default ActionCell;
