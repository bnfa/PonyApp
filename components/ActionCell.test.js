import React from 'react';
import renderer from 'react-test-renderer';
import ActionCell from './ActionCell';

test('ActionCell renders correctly', () => {
  const tree = renderer
    .create(
      <ActionCell
        borders={['north', 'west']}
        isDomokun={true}
        isEndPoint={false}
        isPony={false}
        action="stay"
        onPress={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
