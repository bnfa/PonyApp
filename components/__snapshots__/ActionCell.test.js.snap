// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCell renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#eebff2",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        },
        Object {
          "borderTopColor": "black",
          "borderTopWidth": 1,
        },
        Object {
          "borderLeftColor": "black",
          "borderLeftWidth": 1,
        },
      ]
    }
  >
    <Text>
      D
    </Text>
    <Text>
      P
    </Text>
  </View>
</View>
`;
